docs/vector_search.rst

```diff
@@ -0,0 +1,278 @@
+          "field": "photo_embeddings",
+          "k": 3,
+          "num_candidates": 1000,
+          "query_vector": [0.188423157, 0.246743672, ...]
+        }
+      ]
+    }
+
+Vector Search + Inline Filter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Nrtsearch doesn’t go through the HNSW graph at all. Instead it first filters data using the provided filter. Then it uses a KNN algorithm such as cosine cosine similarity to score returned embeddings. While this option doesn’t traverse through the HNSW graph, it can return very accurate results. It will perform much better when the number of filtered docs to rank is less.
+Example::
+
+    {
+      "indexName": "vector_test",
+      "startHit": 0,
+      "topHits": 10,
+      "timeoutSec": 0,
+      "retrieveFields": [
+        "photo_id",
+        "business_id",
+        "caption"
+      ],
+      "knn": [
+        {
+          "field": "photo_embeddings",
+          "k": 1,
+          "num_candidates": 10,
+       "filter": {
+            "booleanQuery": {
+              "clauses": [
+              {
+                  "occur": "MUST",
+                  "query": {
+                    "termQuery": {
+                      "field": "business_id",
+                      "intValue": 12581436
+                    }
+                  }
+                }
+              ]
+            }
+          },
+          "query_vector": [
+            0.188423157,
+        ....
+       ]
+      }
+     ]
+    }
+
+Here, Nrtsearch will try to find the best matching documents by traversing through the HNSW graph. At the same time it tries to find all the docs matching the top level query clause as well. Then it combines the results using an OR operator. While this approach is using the graph, it’s only recommended for specific use cases, as it won’t provide accurate results. For instance, if one is looking for the most similar photo in a particular business for the given photo, Nrtsearch will find top N photos that are very similar to the given photo across all businesses. It will try to pick the photos that belong to the given business as well. If some or none of the photos from the latter queries are found in the former one, then those photos will be still included in the final results, even though they may not be similar to the photo we are looking for.
+
+An example where this particular use case may make sense is a scenario where one would want to find burger photos for a particular business. The vector search query can find its top burger photo across all businesses. The text search can apply a filter based on business ID and caption field of the document. If there are photos from the same business in the vector search, their score can be boosted using the boost parameter so that when combined with the regular text search results, they get higher score. In this case even if no photos are found from the vector search, the text search can at least show some photos whose caption matches the keyword "burger".
```

Comment:
I think this applies to to the top level filter section below

Reply:
Moved it below